### PR TITLE
Allow setting a Traefik route priority per environment

### PR DESCRIPTION
**Pull request by ptab**
----

 # ⚠️ Could not find ticket corresponding to `allow-defining-traefik-priority-per-environment. Does your branch name follow the correct pattern?

### CODE REVIEW

File: src/mpyl/projects/versioning.py [hunk 1]
```diff
@@ -112,6 +113,22 @@ def upgrade(self, previous_dict: ordereddict) -> ordereddict:
         return previous_dict
 
 
+class ProjectUpgraderOneFour19(Upgrader):
```
Comment (ptab) [inline src/mpyl/projects/versioning.py:116]: I couldn't find any documentation about this process so I had to dig through the project history to find a similar PR and "blindly" apply a similar set of changes. Please give it a careful review as it's likely missing something important.

**Review by MatthijsBon — APPROVED** [review]
Not that I'm of any use in reviewing this MPyL change, but it makes sense to me 😁

File: src/mpyl/schema/project.schema.yml [hunk 2]
```diff
@@ -431,7 +431,27 @@ definitions:
           your ingress for old clients
         type: boolean
       priority:
-        type: number
```
Comment (Jorg88) [inline src/mpyl/schema/project.schema.yml:434]: there is a pre-defined definition for this called `dtapNumberValue`

File: src/mpyl/project.py [hunk 3]
```diff
@@ -361,7 +361,7 @@ class TraefikHost:
     has_swagger: bool
     tls: Optional[TargetProperty[str]]
     whitelists: TargetProperty[list[str]]
-    priority: Optional[int]
+    priority: TargetProperty[int]
```
Comment (koeves) [inline src/mpyl/project.py:364]: I think this should be `Optional[TargetProperty[int]]`

File: src/mpyl/projects/versioning.py (hunk 1 above)
Comment (Jorg88) [reply src/mpyl/projects/versioning.py:116]: I actually found the same problem when I was doing schema change recently and we defined this ticket: https://vandebron.atlassian.net/browse/GUILD-773

**Review by koeves — APPROVED** [review]
Looks good!

You can run the MPyL Test pipeline for this PR to run mpyl on the example projects with your test version: https://jenkins.k8s-serv-backend.vdbinfra.nl/view/MPyL/job/MPyL%20Pipeline%20-%20Test/view/change-requests/

Then it's probably good to test in the monorepo as well. You can run the upgrader script on the monorepo itself and then deploy the projects with the changed priorities in a PR.

Not sure how fast you need this change out, you can also do the release process yourself to try it out :) 
https://vandebron.github.io/mpyl/mpyl.html look for "create a new release"

File: src/mpyl/schema/project.schema.yml (hunk 2 above)
Comment (ptab) [reply src/mpyl/schema/project.schema.yml:434]: Thanks, I wasn't aware of that! I'll apply it in the next commit 👍

File: src/mpyl/project.py (hunk 3 above)
Comment (ptab) [reply src/mpyl/project.py:364]: You're completely right, it should be an optional. I'll apply that change in the next commit 👍

**Comment by ptab** [comment]
This is great info @koeves, thanks for pointing me to that pipeline.

I'm indeed planning to trigger this release myself, to learn how that goes.

File: src/mpyl/projects/versioning.py (hunk 1 above)
Comment (SamTheisens) [reply src/mpyl/projects/versioning.py:116]: @ptab maybe you could add the documentation that you found lacking?

**Comment by SamTheisens** [comment]
> Not related to the changes in this PR: I spent a considerable amount of time trying to figure out why the tests don't pass locally, only to quit and realize that they do pass in CI:
> 
> ```
> ❯ pipenv run validate 
>                                               Validate sourcecode                                               
> ┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
> ┃ Status ┃ Command          ┃ Message                                                                          ┃
> ┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
> │ 💚     │ format           │ All done! ✨ 🍰 ✨                                                               │
> │        │                  │ 161 files would be left unchanged.                                               │
> │ 💚     │ lint             │ --------------------------------------------------------------------             │
> │        │                  │ Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)             │
> │ 💚     │ lint-test        │ --------------------------------------------------------------------             │
> │        │                  │ Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)             │
> │ 💚     │ check-types      │ Success: no issues found in 95 source files                                      │
> │ 💚     │ check-types-test │ Success: no issues found in 58 source files                                      │
> │ 💔     │ test             │ E             - Project-URL: Documentation, https://vandebron.github.io/mpyl     │
> │        │                  │ E             - Project-URL: Source, https://github.com/Vandebron/mpyl...        │
> │        │                  │ E                                                                                │
> │        │                  │ E             ...Full output truncated (36 lines hidden), use '-vv' to show      │
> │        │                  │                                                                                  │
> │        │                  │ tests/test_resources/test_data.py:180: AssertionError                            │
> │        │                  │ =========================== short test summary info ============================ │
> │        │                  │ FAILED tests/cli/commands/test_cli.py::TestCli::test_version_print - Assertio... │
> │        │                  │ FAILED tests/cli/commands/test_cli.py::TestCli::test_verbose_version_print - ... │
> │        │                  │ =================== 2 failed, 144 passed, 4 skipped in 4.96s =================== │
> └────────┴──────────────────┴──────────────────────────────────────────────────────────────────────────────────┘
> ```
> 
> It would be great to be able to validate the project locally.
> 
> UPDATE: I run the same inside a blank Python Docker image and the tests pass, so this is likely caused by some local misconfiguration.

To get more detailed info you can run `pipenv run test`. And for individual test I tend to run them from the IDE.
One difference I have noticed between tests in GHA vs local is that `import mpyl.` instead of `import src.mpyl` does not work locally, as it shouldn't. But works in te pipeline build.

**Review by SamTheisens — APPROVED** [review]

File: releases/notes/1.4.19.md
Comment (Jorg88) [inline releases/notes/1.4.19.md:1]: We're releasing v1.4.19 already now with a fix for the ECR issues, so you should rename this file and update the upgrader logic to 1.4.20. Sorry for the extra work

Comment (ptab) [reply releases/notes/1.4.19.md:1]: No problem, thanks for letting me know! 👍

**Review by Jorg88 — APPROVED** [review]

**Pull request merged**